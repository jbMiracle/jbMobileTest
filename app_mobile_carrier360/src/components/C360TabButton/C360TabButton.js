import React, { Component } from 'react';
import { View, TouchableOpacity, Platform } from 'react-native';
import { connectStyle, Text } from 'native-base';
import styleDef from './C360TabButton.style';
import C360Icon from '../C360Icon/C360Icon';

class C360TabButton extends Component {
  render() {
    const {
      id,
      onPress,
      style,
      tintColor,
      title,
    } = this.props;

    return (
      <TouchableOpacity
        testID={ `bottomnav_${id}` }
        style={ determineStyleType(this.props) }
        onPress={ onPress }
      >
        { renderIcon(this.props) }
        <Text style={ [style.tabTitle, { color: tintColor }] }>{ title }</Text>
      </TouchableOpacity>
    );
  }
}

function determineStyleType(props) {
  const {
    backgroundColor,
    indicator,
    main,
    style,
  } = props;

  const { tabItem, mainItem } = style;

  if (main) {
    return (Platform.OS === 'ios') ? tabItem : mainItem;
  }

  return [tabItem, { backgroundColor, borderTopWidth: 4, borderTopColor: indicator }];
}

function renderIcon(props) {
  const {
    backgroundColor,
    main,
    mainIconColor,
    name,
    style,
  } = props;

  const icon = (
    <C360Icon name={ name } size={ 20 } color={ mainIconColor } />
  );

  if (main) {
    return (
      <View style={ [style.mainButton, { backgroundColor }] }>
        { icon }
      </View>
    );
  }

  return icon;
}

export default connectStyle('C360.TabButton', styleDef)(C360TabButton);
