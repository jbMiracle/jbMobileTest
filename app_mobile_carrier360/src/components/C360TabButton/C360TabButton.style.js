import { tabBarHeight, overallBarHeight } from '../C360TabBar/C360TabBar.style';

export default {
  tabItem: {
    alignItems: 'center',
    height: tabBarHeight,
    justifyContent: 'center',
    paddingTop: 2,
  },
  mainItem: {
    alignItems: 'center',
    height: overallBarHeight,
    justifyContent: 'flex-end',
    paddingBottom: 2,
  },
  tabTitle: {
    fontSize: 11,
    paddingTop: 5,
  },
  tabIcon: {
    paddingBottom: 5,
  },
  mainButton: {
    alignItems: 'center',
    backgroundColor: '#dedede',
    borderRadius: (tabBarHeight - 12) / 2,
    height: tabBarHeight - 12,
    justifyContent: 'center',
    marginTop: -1 * ((tabBarHeight - 12) / 2),
    paddingBottom: 2,
    width: tabBarHeight - 12,
  },
  mainButtonIcon: {
    color: 'white',
  },
};
