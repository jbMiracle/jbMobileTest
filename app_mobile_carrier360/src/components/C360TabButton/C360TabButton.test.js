import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import C360TabButton from './C360TabButton';

it('can render tab button', () => {
  const renderer = new ShallowRenderer();
  const Grey700 = '#292929';
  const Grey400 = '#757575';
  const White = '#fff';
  const props = {
    id: 'buttondId',
    main: false,
    title: 'Example',
    onPress: () => {},
    tintColor: Grey400,
    backgroundColor: White,
    indicator: Grey700,
    mainIconColor: Grey400,
    navigation: {},
  };
  renderer.render(<C360TabButton { ...props } />);

  expect(renderer.getRenderOutput()).toMatchSnapshot();
});
