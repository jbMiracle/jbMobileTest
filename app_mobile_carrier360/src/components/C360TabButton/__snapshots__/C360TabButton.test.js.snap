// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render tab button 1`] = `
<C360TabButton
  backgroundColor="#fff"
  id="buttondId"
  indicator="#292929"
  main={false}
  mainIconColor="#757575"
  navigation={Object {}}
  onPress={[Function]}
  style={
    Object {
      "mainButton": Object {
        "alignItems": "center",
        "backgroundColor": "#dedede",
        "borderRadius": 24,
        "height": 48,
        "justifyContent": "center",
        "marginTop": -24,
        "paddingBottom": 2,
        "width": 48,
      },
      "mainButtonIcon": Object {
        "color": "white",
      },
      "mainItem": Object {
        "alignItems": "center",
        "height": 60,
        "justifyContent": "flex-end",
        "paddingBottom": 2,
      },
      "tabIcon": Object {
        "paddingBottom": 5,
      },
      "tabItem": Object {
        "alignItems": "center",
        "height": 60,
        "justifyContent": "center",
        "paddingTop": 2,
      },
      "tabTitle": Object {
        "fontSize": 11,
        "paddingTop": 5,
      },
    }
  }
  tintColor="#757575"
  title="Example"
  virtual={undefined}
/>
`;
